app/admin/views/items.py

```diff
@@ -64,6 +62,7 @@ class ItemsView(BaseView):
     ]
 
     form_edit_rules = [
+        rules.FieldSet(['storage_id'], 'Storage'),
```

Comment:
You must check that the new storage belongs to the same user as the old storage.

Reply:
Added check in `on_model_change` but found out that it doesn't work as intended. `form` and `model` represented the same data so I had to redefine `edit model`. Moved `on_model_change` upper so it would be called before `model` was updated